src/data/store.rs

```diff
@@ -79,13 +80,37 @@ impl Store {
         store.downloads.load_downloaded_files();
         store.downloads.load_pending_downloads();
 
+        store.chats.load_chats();
+
         store.search.load_featured_models();
         store
     }
 
-    pub fn load_model(&mut self, file: &File) {
-        if self.chats.load_model(file).is_ok() {
-            self.preferences.set_current_chat_model(file.id.clone());
+    // TODO: This wrappers feel weird.
+    pub fn create_new_chat(&mut self, file: &File, set_as_current: bool) {
+        if self.chats.create_new_chat(file, set_as_current).is_ok() {
+            // TODO: This breaks, FIX THE PREFERENCES!!!
+            // self.preferences.set_current_chat_model(file.id.clone());
+        }
```

Comment:
I would leave the existing `load_model` function like this:

```rust
    pub fn load_model(&mut self, file: &File, set_as_current: bool) {
        // We are creating a new chat because there is no other way to start a new conversation
        // with a model. This is a temporary solution until we have a better way to start a chat

        if self.chats.load_model(file).is_ok()
            && self.chats.create_new_chat(file, set_as_current).is_ok()
        {
            self.preferences.set_current_chat_model(file.id.clone());
        }
    }
```

As stated by the comment, the `create_new_chat` will be shortly removed with the PR I'm working on.

By the way, the preference line is working in my local setup.